samples/place-nearby-search/index.ts

```diff
@@ -5,72 +5,112 @@
-    //@ts-ignore
     const { Place, SearchNearbyRankPreference } = await google.maps.importLibrary('places') as google.maps.PlacesLibrary;
     const { AdvancedMarkerElement } = await google.maps.importLibrary("marker") as google.maps.MarkerLibrary;
+    const { spherical } = await google.maps.importLibrary('geometry') as google.maps.GeometryLibrary;
     // [START maps_place_nearby_search_request]
-
-    // Restrict within the map viewport.
-    let center = new google.maps.LatLng(52.369358, 4.889258);
+    // Get bounds and radius to constrain search.
+    
+    center = map.getCenter();
+    let bounds = map.getBounds();
+    let ne = bounds.getNorthEast();
+    let sw = bounds.getSouthWest();
+    let diameter = spherical.computeDistanceBetween(ne, sw);
+    let radius = Math.min((diameter / 2 ), 50000); // Radius cannot be more than 50000.
 
     const request = {
         // required parameters
-        fields: ['displayName', 'location', 'businessStatus'],
+        fields: ['displayName', 'location', 'formattedAddress', 'googleMapsLinks'],
         locationRestriction: {
-            center: center,
-            radius: 500, 
+            center,
+            radius,
         },
         // optional parameters
-        includedPrimaryTypes: ['restaurant'],
+        includedPrimaryTypes: [typeSelect.value],
         maxResultCount: 5,
         rankPreference: SearchNearbyRankPreference.POPULARITY,
         language: 'en-US',
         region: 'us',
     };
 
-    //@ts-ignore
     const { places } = await Place.searchNearby(request);
     // [END maps_place_nearby_search_request]
 
     if (places.length) {
-        console.log(places);
-
         const { LatLngBounds } = await google.maps.importLibrary("core") as google.maps.CoreLibrary;
         const bounds = new LatLngBounds();
 
+        // First remove all existing markers.
+        for (const id in markers) {
+            markers[id].map = null;
+        };
+        markers = {};
+        
         // Loop through and get all the results.
-        places.forEach((place) => {
-            const markerView = new AdvancedMarkerElement({
+        places.forEach(place => {
+            const marker = new AdvancedMarkerElement({
                 map,
                 position: place.location,
                 title: place.displayName,
             });
+            markers[place.id] = marker;
+
+           //@ts-expect-error
+           let content = `${place.formattedAddress}<br/>${place.id}<br/><a href="${place.googleMapsLinks.placeURI}">View Details on Google Maps</a>`;
 
-            bounds.extend(place.location as google.maps.LatLng);
-            console.log(place);
+            marker.addListener('gmp-click', () => {
+                map.panTo(place.location);
+                updateInfoWindow(place.displayName, content, marker);
+            });
+
+            if (place.location != null) {
+                bounds.extend(place.location);
+            }
```

Comment:
thoughts on moving this to the top of the loop, and inverting?

```
if (!place.location) return;
bounds.extend(place.location);
```

Reply:
Done.